src/test/mir-opt/basic_assignment.rs

```diff
@@ -13,40 +15,10 @@ fn main() {
     // assignment:
     nodrop_y = nodrop_x;
 
-    let drop_x : Option<Box<u32>> = None;
+    let drop_x: Option<Box<u32>> = None;
     let drop_y;
 
     // Since the type of `drop_y` has drop, we generate a `replace`
     // terminator:
     drop_y = drop_x;
 }
-
-// END RUST SOURCE
-// START rustc.main.SimplifyCfg-initial.after.mir
-//    bb0: {
-//        StorageLive(_1);
-//        _1 = const false;
-//        FakeRead(ForLet, _1);
-//        StorageLive(_2);
-//        StorageLive(_3);
-//        _3 = _1;
-//        _2 = move _3;
-//        StorageDead(_3);
-//        StorageLive(_4);
-//        _4 = std::option::Option::<std::boxed::Box<u32>>::None;
-//        FakeRead(ForLet, _4);
-//        AscribeUserType(_4, o, UserTypeProjection { base: UserType(1), projs: [] });
-//        StorageLive(_5);
-//        StorageLive(_6);
-//        _6 = move _4;
-//        replace(_5 <- move _6) -> [return: bb2, unwind: bb5];
-//    }
-//    ...
-//    bb2: {
-//        drop(_6) -> [return: bb6, unwind: bb4];
-//    }
-//    ...
-//    bb5 (cleanup): {
-//        drop(_6) -> bb4;
-//    }
-// END rustc.main.SimplifyCfg-initial.after.mir
```

Comment:
Compared to the emitted file that we are getting now, this one here focuses better on what really should be tested. How do we prevent people from just `--bless`ing any changes they make, which would totally miss important regressions?

Basically, this lacks the equivalent of `//~ ERROR` in ui tests, where even `--bless` will not accept any kind of change.

Reply:
Well... the current system still allowed you to just copy paste the MIR over the existing mir-opt expected MIR, it was just more cumbersome to do so. I don't think the effort of changing mir-opt tests affects whether we miss regressions or not, it just made doing changes to MIR very annoying.

The current system still exists, so you can have both in one file, though I would have hoped we could eliminate the current system at some point.